core/src/com/mygdx/game/models/Command.java

```diff
@@ -1,14 +1,93 @@
 package com.mygdx.game.models;
 
-public interface Command {
+/**
+ * Class to represent a command
+ */
+public class Command {
+    /* Type of action the command represents */
+    private String actionType;
 
-    /**
-     * Get the action associated with the command
-     */
-    String getAction();
+    /* Acting unit's ID */
+    private int unitId;
 
-    /**
-     * Get position that the action is directed at
-     */
-    Position getPosition();
+    /* Target unit's ID */
+    private int targetUnitId;
+
+    /* Position action is taken from */
+    private Position startPosition;
+
+    /* Targeted position */
+    private Position targetPosition;
+
+    /* The number of turns the command has been active */
+    private int passedDuration;
+
+    /* How long the command takes */
+    private int totalDuration;
```

Comment:
this makes things more complicated. If individual commands can have durations greater than one turn then orders is wrong. Since you wouldn't always increment the order if the command lasts more than one turn.

Reply:
Deleted. Will find another way to implement this later. Example of a better way to implement this:
class BuildOrder {
    int progress;
    void incrementProgress() {
        progress += 1;
    }
}

class build {
    void do() {
        incrementProgress();
    }
}